lib/test_cases/function/pull_request_file.js

```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
+                      }
+                  }
+                );
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Check duplicate code in switch case.
+     */
+    function pullRequestFileCheckSwitchDuplicateCode(prFile) {
+        let result = [];
+        let caseStart = 0;
+        let case_statements = [];
+        let is_duplicate_code = 0;
+        if (prFile.type === 'php' || prFile.type === 'module' || prFile.type === 'theme') {
+            let targetPrLine = undefined;
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("case \'[a-zA-Z0-9]+\'\:");
+                    let content = currentPrLine.content;
+                    if (content.includes("switch")) {
+                        targetPrLine = currentPrLine
+                        case_statements = [];
+                    }
+                    if (content.includes("break;")) {
+                        caseStart = 0;
+                    }
+                    if (caseStart) {
+                        if (case_statements.indexOf(content.trim()) > -1) {
+                            is_duplicate_code = 1;
+                        } else {
+                            case_statements.push(content.trim())
+                        }
+                    }
+                    if (pattern.test(content)) {
+                        caseStart = 1;
+                    }
+                }
+                if (is_duplicate_code) {
+                    result.push(
+                      {
+                          line: targetPrLine,
+                          test: {
+                              category: ['best_practices'],
+                              title: 'Avoid code duplicacy in switch cases',
+                              description: 'Avoid duplicate code statements in switch case.'
+                          }
+                      }
+                    );
+                    is_duplicate_code = 0;
+                    targetPrLine = undefined;
+                }
+            });
+        }
+
+        return result;
+    }
+
+    /**
+     * Check for name tags in subscriber service.
+     */
+    function pullRequestFileCheckSubscriberServiceTags(prFile) {
+        let result = [];
+        if (prFile.type === 'services.yml') {
+            let targetPrLine = undefined;
+            let is_pattern_present = 0;
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("[a-z]+_subscriber\:");
+                    let content = currentPrLine.content;
+                    if (pattern.test(content)) {
+                        is_pattern_present = 1;
+                        targetPrLine = currentPrLine
+                    }
+                    if (is_pattern_present) {
+                        if (content.includes("tags:")) {
```

Comment:
content - is of currentPrLine while tags: can be present on next line, are you sure this working?